addons/binding/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/discovery/MagentaTVDiscoveryParticipant.java

```diff
@@ -0,0 +1,153 @@
+                        device.getDetails().getFriendlyName(), device.getIdentity().getUdn().getIdentifierString(),
+                        device.getDetails().getModelDetails().getModelName(),
+                        device.getDetails().getModelDetails().getModelNumber());
+                if (handlerFactory == null) {
+                    logger.error("handlerFactory not yet initialized!");
+                    return null;
+                }
+
+                Map<String, Object> properties = new HashMap<>();
+                properties.put(PROPERTY_VENDOR,
+                        VENDOR + "(" + device.getDetails().getManufacturerDetails().getManufacturer() + ")");
+                String model = device.getDetails().getModelDetails().getModelName();
+                properties.put(PROPERTY_MODEL_ID, device.getDetails().getModelDetails().getModelName().toUpperCase());
+                properties.put(PROPERTY_HARDWARE_VERSION, device.getDetails().getModelDetails().getModelNumber());
+                properties.put(PROPERTY_SERIAL_NUMBER, device.getDetails().getSerialNumber());
+                properties.put(PROPERTY_UDN, device.getIdentity().getUdn().getIdentifierString().toUpperCase());
+                String descriptorURL = device.getIdentity().getDescriptorURL().toString();
+                properties.put(PROPERTY_IP, StringUtils.substringBetween(descriptorURL, "http://", ":"));
+                String port = StringUtils.substringBefore(StringUtils.substringAfterLast(descriptorURL, ":"), "/");
+                properties.put(PROPERTY_PORT, port);
+                properties.put(PROPERTY_DESC_URL, StringUtils.substringAfterLast(descriptorURL, ":" + port));
+
+                String hex = device.getIdentity().getUdn().getIdentifierString()
+                        .substring(device.getIdentity().getUdn().getIdentifierString().length() - 12);
+                String mac = hex.substring(0, 2) + ":" + hex.substring(2, 4) + ":" + hex.substring(4, 6) + ":"
+                        + hex.substring(6, 8) + ":" + hex.substring(8, 10) + ":" + hex.substring(10, 12);
+                properties.put(PROPERTY_MAC_ADDRESS, mac);
+
+                logger.debug("Create Thing for device '{}' with UDN '{}, Model'{}'",
+                        device.getDetails().getFriendlyName(), device.getIdentity().getUdn().getIdentifierString(),
+                        device.getDetails().getModelDetails().getModelName());
+                // TO-DO: Check if thing with this name already exist
+                result = DiscoveryResultBuilder.create(uid).withProperties(properties)
+                        .withLabel(device.getDetails().getFriendlyName()).withRepresentationProperty(PROPERTY_UDN)
+                        .build();
+                handlerFactory.deviceDiscoverd(properties);
+            }
+        } catch (Exception e) {
+            logger.error("Unable to create thing: {} ({})", e.getMessage(), e.getClass());
```

Comment:
logger.error
I also do not really see where an exception can be thrown in this process. That's the reason why it usually is not a good idea to catch a generic Exception. For one: The framework can handle it if this method throws. And also: The original exception and stacktrace is hidden now.

Reply:
I had exceptions from DiscoveryResultBuilder.create(), catching this and adding a specific error message makes it way more specific and easier to find than by the frameworks stack trace. As a developer I know the technical flow in the code, a user can't deal with the stack trace information.
**-> won't agree**

Reply:
DiscoveryResultBuilder.create can't throw if you put in an existing ThingUID. 

Your way of dealing with the situation is like wrapping every method with a try catch because a NPE could happen. That's really brute force. Try to find the problem why your ThingUID was broken and fix it and it will never happen again. The entire try/catch would be useless and just eats up stack space.